src/biokbase/narrative/tests/test_jobcomm.py

```diff
@@ -293,37 +271,41 @@ def test_start_job_status_loop__cell_ids(self):
 
                 self.jc.start_job_status_loop(init_jobs=True, cell_list=combo)
                 msg = self.jc._comm.last_message
-                self.assertEqual(
-                    {
-                        "msg_type": "job_status_all",
-                        "content": get_test_job_states(
-                            EXP_ALL_STATE_IDS
-                        ),  # consult version history for when this was exp_job_ids
-                    },
-                    msg["data"],
-                )
 
-                self.assertTrue(self.jc._running_lookup_loop)
-                self.assertTrue(self.jc._lookup_timer)
+                exp_job_ids = [
+                    job_id
+                    for cell_id, job_ids in cell_2_jobs.items()
+                    for job_id in job_ids
+                    if cell_id in combo and not JOBS_TERMINALITY[job_id]
+                ]
+                exp_msg = {
+                    "msg_type": "job_status_all",
+                    "content": get_test_job_states(exp_job_ids),
+                }
+                self.assertEqual(exp_msg, msg["data"])
 
-                self.jc.stop_job_status_loop()
-                self.assertFalse(self.jc._running_lookup_loop)
-                self.assertIsNone(self.jc._lookup_timer)
+                if len(exp_job_ids):
```

Comment:
This batch of jobs gets generated - it's possible to not have any job ids. If not, then the loop shouldn't run, and we don't care about it anyway.